src/Grant/JwtGrant.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare(strict_types=1);
+
+namespace DocusignBundle\Grant;
+
+use Lcobucci\JWT\Builder;
+use Lcobucci\JWT\Signer\Key;
+use Lcobucci\JWT\Signer\Rsa\Sha256;
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Contracts\HttpClient\Exception\ExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Vincent Chalamon <vincentchalamon@gmail.com>
+ */
+final class JwtGrant implements GrantInterface
+{
+    private $client;
+    private $privateKey;
+    private $integrationKey;
+    private $userGuid;
+    private $apiUri;
+    private $accountApiUri;
+
+    public function __construct(
+        string $privateKey,
+        string $integrationKey,
+        string $userGuid,
+        string $apiURI,
+        string $accountApiUri,
+        HttpClientInterface $client = null
+    ) {
+        $this->client = $client ?: HttpClient::create();
+        $this->privateKey = $privateKey;
+        $this->integrationKey = $integrationKey;
+        $this->userGuid = $userGuid;
+        $this->apiUri = $apiURI;
+        $this->accountApiUri = $accountApiUri;
+    }
+
+    public function __invoke(): string
+    {
+        $time = time();
+        $token = (new Builder())->issuedBy($this->integrationKey) // iss
+            ->relatedTo($this->userGuid) // sub
+            ->issuedAt($time) // iat
+            ->expiresAt($time + 3600) // exp
```

Comment:
IMO this should be a configuration option. In some cases I might need to reduce this value.